### ISSUE TITLE
Allow to disable injection or extraction on propagators

### ISSUE THREAD

**Issue by marcinjahn** [open-telemetry/opentelemetry-specification#3153]
**What are you trying to achieve?**

OpenTelemetry has the Propagators concept and implementations (e.g. W3C or B3 Propagator). By default, the W3C propagator is utilized. The other propagators are used mostly to keep compatibility with other, often older services.

The issue is that enabling a non-W3C propagator always implies that our service will not only be able to understand incoming non-W3C traces, but it will also emit them further on. That might be non-desirable. 

A simple example would be a case where 3 services are present:

A --> B --> C

Service A uses B3 formatted tracing. It sends a request to B. B receives the request and wants to keep the traceId that A sent to it. B makes a request to C. C is a modern service that uses W3C propagation format. It does not need the B3 trace headers. More than that, we don't want to encourage any downstream service to use the B3-formatted tracing headers. By sending these headers to other services (like C), we're giving them a message that we do intend to support B3. In reality, we support it only because A does. As soon as A drops this format and moves to W3C, we'd remove the propagator.

**What did you expect to see?**

Based on the example above, I'd expect to be able to disable the injection on selected propagators. However, I think it would make more sense to allow for both the extraction, and injection to be either enabled, or disabled.

I'm not sure if proposing implementation is the right thing to do at this point, but I'd imagine that every propagator could accept an enum, such as:

```cs
enum PropagationOptions 
{
   InjectAndExtract,
   OnlyInject,
   OnlyExtract,
   Disable // not sure if that one is needed, since it would be a kind of NoOp propagator
}
```

**Additional context.**

Originally, I have submitted this PR to the JS repo: https://github.com/open-telemetry/opentelemetry-js/pull/3568
I was advised by @dyladan to introduce the idea here instead.

**Comment by MrAlias** [comment]
I'm not sure how the JS propagator interface is structured, but in Go it is possible for a user to provide their own propagator that handles any extract/inject logic they want. For example they can wrap the OTel supported B3 and W3C tracecontext propagators and for extract/inject with either of these. It could even be built in a dynamically configured way.

I do not think this is a needed addition to the specification.

**Comment by marcinjahn** [comment]
Yeah, I believe this is possible in any/most of the languages that OT supports, including JS of course.

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> MEMBER MrAlias replied that users can already wrap the built-in B3/W3C propagators to control extract/inject and "I do not think this is a needed addition to the specification" (comment 1409202158, thumbs-up'd). The author (marcinjahn) conceded this is possible in most languages and did not push back; the originating JS PR #3568 is closed. Discussion fizzled in Jan 2023 with maintainer signaling no intent to act.

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> Verified: MEMBER MrAlias said users can wrap built-in propagators themselves and "I do not think this is a needed addition to the specification" (thumbs-up'd); author marcinjahn conceded it's possible in most languages and did not push back; originating JS PR #3568 confirmed CLOSED; no activity since 2023-01-30.

Claude evaluated this issue with medium confidence that it can be closed.

Claude has classified this issue as stale.

The intention of this comment is to make it easier for a human being to review this issue before making a decision.

**Comment by svrnm** [comment]
@marcinjahn I am closing this issue now based on the feedback given, if you think this is incorrect, please reopen or create a new issue, thanks.
